storage/src/main/java/org/apache/kafka/storage/internals/log/LogSegment.java

```diff
@@ -253,17 +253,33 @@ public void append(long largestOffset,
             // append the messages
             long appendedBytes = log.append(records);
             LOGGER.trace("Appended {} to {} at end offset {}", appendedBytes, log.file(), largestOffset);
-            // Update the in memory max timestamp and corresponding offset.
-            if (largestTimestampMs > maxTimestampSoFar()) {
```

Comment:
With this change, we can remove both `largestTimestampMs` and `shallowOffsetOfMaxTimestamp`. We probably can also remove `LogAppendInfo.shallowOffsetOfMaxTimestamp`.

Reply:
Yes, removed `largestTimestampMs` and `shallowOffsetOfMaxTimestamp` in `LogSegment#append`. Also, removed `LogAppendInfo.shallowOffsetOfMaxTimestamp` because it's not referenced anymore.